From 3ea6c53a44f5d7a25b0274ad384d445739f85203 Mon Sep 17 00:00:00 2001
From: Sumit Bose <sbose@redhat.com>
Date: Mon, 3 Dec 2012 11:45:49 +0100
Subject: [PATCH] Restart sssd after authconfig update

Recent versions of authconfig do not restart sssd if only the
--enablesssd and --enablesssdauth options are used. To make sure sssd is
running after ipa-server-install is run this patch add an unconditional
restart of sssd after authconfig is run during the installation.

Since there already is some logic trying to determine if sssd needs to
be restarted or stopped if freeipa in uninstalled no changes are needed
here.

Fixes https://fedorahosted.org/freeipa/ticket/3267
---
 ipa-client/ipa-install/ipa-client-install | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/ipa-client/ipa-install/ipa-client-install b/ipa-client/ipa-install/ipa-client-install
index 190efb183d8c96e2c9665cf51d5346dc1111ae24..612bdc658870fa680f8d7c5b30d8111a6a316c66 100755
--- a/ipa-client/ipa-install/ipa-client-install
+++ b/ipa-client/ipa-install/ipa-client-install
@@ -1822,6 +1822,12 @@ def install(options, env, fstore, statestore):
         auth_config.add_option("update")
         auth_config.execute()
         root_logger.info("%s", message)
+        if options.sssd:
+            sssd = ipaservices.service('sssd')
+            try:
+                sssd.restart()
+            except CalledProcessError:
+                root_logger.warning("SSSD service restart was unsuccessful.")
 
         if not options.sssd:
             #Modify pam to add pam_krb5 only when sssd is not in use
-- 
1.7.11.7

